cmd/oras/login.go

```diff
@@ -74,100 +60,70 @@ Example - Login with insecure registry from command line:
   oras login --insecure localhost:5000
 `,
 		Args: cobra.ExactArgs(1),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return preRunLogin(opts)
+		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			opts.hostname = args[0]
+			opts.Hostname = args[0]
 			return runLogin(opts)
 		},
 	}
-
-	cmd.Flags().BoolVarP(&opts.debug, "debug", "d", false, "debug mode")
-	cmd.Flags().StringArrayVarP(&opts.configs, "config", "c", nil, "auth config path")
-	cmd.Flags().StringVarP(&opts.username, "username", "u", "", "registry username")
-	cmd.Flags().StringVarP(&opts.password, "password", "p", "", "registry password or identity token")
-	cmd.Flags().BoolVarP(&opts.fromStdin, "password-stdin", "", false, "read password or identity token from stdin")
-	cmd.Flags().BoolVarP(&opts.insecure, "insecure", "k", false, "allow connections to SSL registry without certs")
-	cmd.Flags().StringVarP(&opts.caFilePath, "ca-file", "", "", "server certificate authority file for the remote registry")
-	cmd.Flags().BoolVarP(&opts.plainHTTP, "plain-http", "", false, "allow insecure connections to registry without SSL")
-	cmd.Flags().BoolVarP(&opts.verbose, "verbose", "v", false, "verbose output")
+	option.ApplyFlags(&opts, cmd.Flags())
 	return cmd
 }
 
-func runLogin(opts loginOptions) (err error) {
-	var logLevel logrus.Level
-	if opts.debug {
-		logLevel = logrus.DebugLevel
-	} else if opts.verbose {
-		logLevel = logrus.InfoLevel
-	} else {
-		logLevel = logrus.WarnLevel
-	}
-	ctx, _ := trace.WithLoggerLevel(context.Background(), logLevel)
-
-	// Prepare auth client
-	store, err := credential.NewStore(opts.configs...)
-	if err != nil {
+func preRunLogin(opts loginOptions) (err error) {
+	if err := opts.ReadPassword(); err != nil {
 		return err
 	}
```

Comment:
Only this part should be in the `PreRun` and `oras pull` and `oras push` may have the same `PreRun`. Other parts are the main body of the login process.